### ISSUE TITLE
Row panel error after refactor of determining panel type.

### ISSUE THREAD

**Issue by joshlreese** [grafana/grafonnet#49]
It appears that a few days ago when #43 was introduced, calling the `panel.row.new` function results in a missing argument error. I've included an example to reproduce this and the output below.

**Code**

```jsonnet
local g = import 'github.com/grafana/grafonnet/gen/grafonnet-latest/main.libsonnet';

g.panel.row.new('Test')
```

**Result**

```shell
RUNTIME ERROR: Missing argument: value
	vendor/github.com/grafana/grafonnet/grafonnet-base/veneer/panel.libsonnet:10:7-22	function <anonymous>
	dashboard.jsonnet:3:1-27	$
	During evaluation
```

**Expected result**

```json
{
   "datasource": {
      "type": "datasource",
      "uid": "-- Mixed --"
   },
   "title": "Test",
   "type": "row"
}
```

I've found that the following patch addresses the issue, though I'm not quite up to speed on what's best for grafonnet or jsonnet in general, and it's my understanding that #43 intended to derive this information automatically:

```diff
diff --git a/grafonnet-base/main.libsonnet b/grafonnet-base/main.libsonnet
index d797051..7db400a 100644
--- a/grafonnet-base/main.libsonnet
+++ b/grafonnet-base/main.libsonnet
@@ -104,6 +104,11 @@ local veneer = import './veneer/main.libsonnet';
       panel+: {
         row:
           root.coreLib.new(rowPanel)
+          + {
+              withType(): {
+                type: 'row',
+              }
+            }
           + root.packageDocMixin(version, 'row', 'panel.')
           + veneer.panel('row'),
       },
```

I'm aware that this is an experimental library, and look forward to leveraging it and contributing where we can. Thanks for working on it!

**Comment by MartinRoman7** [comment]
Hi Grafonnet team!

Today we tested the grafonnet-latest package with the fix #51, and the issue still appears to be happening. 

```bash
RUNTIME ERROR: Missing argument: value
	lib/dependencies/github.com/grafana/grafonnet/grafonnet-base/veneer/panel.libsonnet:10:7-22
```

**Comment by MartinRoman7** [comment]
Thanks! Now it's working. :+1:
